examples/voting/src/lib.rs

```diff
@@ -0,0 +1,415 @@
+type VoteIndex = u32;
+type VoteCount = u32;
+
+/// The parameter type for the contract function `vote`.
+/// Takes a vote_index that the account wants to vote for.
+#[derive(Deserial, SchemaType)]
+struct VoteParameter {
+    /// Voting option index to vote for.
+    vote_index: VoteIndex,
+}
+
+/// The parameter type for the contract function `init`.
+/// Takes a description, the voting options, and the end_time to start the
+/// election.
+#[derive(Deserial, SchemaType)]
+struct InitParameter {
+    /// The description of the election.
+    description: String,
+    /// A vector of all voting options.
+    options:     Vec<VotingOption>,
+    /// The time when the election ends.
+    end_time:    Timestamp,
+}
+
+/// The return_value type of the contract function `view`.
+/// Returns a description, the end_time, the voting options as a vector, and the
+/// number of voting options of the current election.
+#[derive(Serial, Deserial, SchemaType)]
+struct VotingView {
+    /// The description of the election.
+    description: String,
+    /// The time when the election ends.
+    end_time:    Timestamp,
+    /// A vector of all voting options.
+    options:     Vec<VotingOption>,
+    /// The number of voting options.
+    num_options: u32,
+}
+
+/// The contract state
+#[derive(Serial, DeserialWithState, StateClone)]
+#[concordium(state_parameter = "S")]
+struct State<S> {
+    /// The description of the election.
+    description: String,
+    /// The map connects a voter to the index of the voted-for voting option.
+    ballots:     StateMap<AccountAddress, VoteIndex, S>,
+    /// The map connects the index of a voting option to the number of votes
+    /// it received so far.
+    tally:       StateMap<VoteIndex, VoteCount, S>,
+    /// The time when the election ends.
+    end_time:    Timestamp,
+    /// A vector of all voting options.
+    options:     Vec<VotingOption>,
+    /// The number of voting options.
+    num_options: u32,
+}
+
+/// The different errors that the `vote` function can produce.
+#[derive(Reject, Serial, PartialEq, Eq, Debug, SchemaType)]
+enum VotingError {
+    /// Raised when parsing the parameter failed.
+    #[from(ParseError)]
+    ParsingFailed,
+    /// Raised when the log is full.
+    LogFull,
+    /// Raised when the log is malformed.
+    LogMalformed,
+    /// Raised when the vote is placed after the election has ended.
+    VotingFinished,
+    /// Raised when voting for a voting index that does not exist.
+    InvalidVoteIndex,
+    /// Raised when a smart contract tries to participate in the election. Only
+    /// accounts are allowed to vote.
+    ContractVoter,
+}
+
+/// Mapping the logging errors to VotingError.
+impl From<LogError> for VotingError {
+    fn from(le: LogError) -> Self {
+        match le {
+            LogError::Full => Self::LogFull,
+            LogError::Malformed => Self::LogMalformed,
+        }
+    }
+}
+
+type VotingResult<T> = Result<T, VotingError>;
```

Comment:
Add doc comment similar to what we do elsewhere. Something like "this is an alias that fixes the error type to `VotingError`"

Reply:
A custom alias type for the `Result` type with the error type set to `VotingError`.